trio/_core/_thread_cache.py

```diff
@@ -1,7 +1,48 @@
 from threading import Thread, Lock
 import outcome
+import sys
+import ctypes
+import ctypes.util
 from itertools import count
 
+from typing import Callable, Optional, Tuple
+from functools import partial
+
+
+def get_os_thread_name_func() -> Optional[Callable[[Optional[int], str], None]]:
+    def namefunc(setname: Callable[[int, bytes], int], ident: Optional[int], name: str):
+        if ident is not None:
+            setname(ident, bytes(name[:15], "ascii", "replace"))
```

Comment:
Let's add a helper function:

```python
def _to_os_thread_name(name: str) -> bytes:
    if name.startswith("Thread for "):  # str.removeprefix is new in 3.9
        name = name.replace("Thread for ", "")
    return name.encode("ascii", errors="replace")[:16]  # 15 chars + implicit trailing \x00
```

Reply:
btw `[:16]` doesn't work - it seems ctypes adds the null byte on it's own so it fails on anything more than `[:15]`